### PR TITLE
open ot proxy

### PR DESCRIPTION
**Pull request by huguesBouvier**
otproxy is always closed from the network. This prevent security patches from being applied.

With this PR, otproxy will be open by default and locked only when performing the tests. This way, most of the time OT proxy will be open to new security patches.

Lock
Checked that the VM is locked in portal

Unlock
Checked that the VM is unlocked in portal.

https://dev.azure.com/msazure/One/_build/results?buildId=56491150&view=logs&j=c27c4ff5-2c93-5622-6650-0f5d00fc53b7&t=84dfacbf-177f-5115-f3f0-0f5e9b223605

### CODE REVIEW

File: builds/e2e/isa-95-smoke-test.yaml [hunk 1]
```diff
@@ -203,6 +203,10 @@ stages:
         parameters:
           agentName: $(deviceLvl4AgentName)
           lvl: 4
+      - template: templates/nested-isa95-lock.yaml      
+        parameters:
+          agentName: "otproxy"
```
Comment (varunpuranik) [inline builds/e2e/isa-95-smoke-test.yaml:208]: Is this supposed to be hardcoded?

**Review by varunpuranik — APPROVED** [review]

File: builds/e2e/isa-95-smoke-test.yaml [hunk 2]
```diff
@@ -251,7 +255,8 @@ stages:
       deviceLvl4AgentName: $[ stageDependencies.RunISA95Tests.SetupVM_level4.outputs['exportAgentName.agentName'] ]       
       deviceLvl3AgentName: $[ stageDependencies.RunISA95Tests.SetupVM_level3.outputs['exportAgentName.agentName'] ]  
       deviceLvl5NSGName: $[ stageDependencies.RunISA95Tests.Run_ISA95_test.outputs['isa95_lock_lvl5.nsgName'] ]         
-      deviceLvl4NSGName: $[ stageDependencies.RunISA95Tests.Run_ISA95_test.outputs['isa95_lock_lvl4.nsgName'] ]       
+      deviceLvl4NSGName: $[ stageDependencies.RunISA95Tests.Run_ISA95_test.outputs['isa95_lock_lvl4.nsgName'] ] 
+      deviceLvl35NSGName: $[ stageDependencies.RunISA95Tests.Run_ISA95_test.outputs['isa95_lock_lvl35.nsgName'] ]
```
Comment (damonbarry) [inline builds/e2e/isa-95-smoke-test.yaml:259]: Does this mean we already had all the logic necessary to close off otproxy from the public internet and open it back up again (we just weren't applying it)?

**Review by damonbarry — COMMENTED** [review]
It would be great to have a better description of the problem this PR is solving, and then how this PR solves the problem.

File: builds/e2e/isa-95-smoke-test.yaml (hunk 2 above)
Comment (huguesBouvier) [reply builds/e2e/isa-95-smoke-test.yaml:259]: Yes, we did only use it to open the l3 l4 and l5

File: builds/e2e/isa-95-smoke-test.yaml (hunk 1 above)
Comment (huguesBouvier) [reply builds/e2e/isa-95-smoke-test.yaml]: Yes but I can do better than this. I moved it at the top with other hardcoded name.

**Comment by huguesBouvier** [comment]
> It would be great to have a better description of the problem this PR is solving, and then how this PR solves the problem.

Done

**Review by damonbarry — APPROVED** [review]

**Pull request merged**